### PR TITLE
chore: container script run through CLI args

### LINKED ISSUE #6598: Run container using CLI arguments instead of environment variables
Currently, the container is started using an environment variable to decide if server or celery instance will be run. It makes it difficult to modify the image for quick edits when running. 

https://github.com/fossasia/open-event-server/blob/b3a8c3305ec232f3a0a6b30c5961ca5a43e26df5/scripts/container_start.sh#L2-L9
https://github.com/fossasia/open-event-server/blob/b3a8c3305ec232f3a0a6b30c5961ca5a43e26df5/scripts/container_start.sh#L21-L23

Instead it should be run via CLI arguments. Like

`./container_start.sh run api` or `./container_start.sh run celery`

### PR DESCRIPTION
**Pull request by codedsun**
Fixes #6598 

#### Short description of what this resolves:
Container script now can be run by CLI args instead of envt variables

#### Changes proposed in this pull request:

- Accessed CLI Args

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by codedsun** [comment]
@iamareebjamal Review

**Comment by iamareebjamal** [comment]
docker-compose.yml needs to be updated as well

**Comment by codedsun** [comment]
@iamareebjamal Their is a dockerfile for same which runs this script command

```
CMD ["sh", "scripts/container_start.sh"]
```

how do i know what to run celery or app?

**Comment by iamareebjamal** [comment]
Read what I have asked to modify and what you are reading

https://www.diffchecker.com/SOPSN5cp

**Comment by codedsun** [comment]
```
celery:
    <<: *defaults
    container_name: opev-celery
    environment:
     <<: *environment-vars
      DEPLOYMENT: celery
      C_FORCE_ROOT: "true"

```

Does the environment-vars line need to be removed?

**Comment by iamareebjamal** [comment]
No

**Comment by codedsun** [comment]
@iamareebjamal used the command variable to run the script

**Comment by codedsun** [comment]
@iamareebjamal Will the command variable not be used? Please guide me

**Comment by iamareebjamal** [comment]
> Does the environment-vars line need to be removed?
No

Then why did you remove it?

**Comment by iamareebjamal** [comment]
Run `docker-compose up -d` and `docker ps` 

If server, celery, postgres and redis are running and site is working, then ask for review

**Comment by codedsun** [comment]
@iamareebjamal They are running without doing anything in docker-compose, maybe you meant to remove the deployment environment variable passed in docker-compose, am I right now?

**Comment by iamareebjamal** [comment]
Still won't work. 

https://github.com/fossasia/open-event-server/pull/6599#issuecomment-554633517

**Comment by iamareebjamal** [comment]
> They are running without doing anything in docker-compose

Dude, they are running locally, you need to test that it works in docker. Where the container_start.sh script is used.

**Comment by codedsun** [comment]
> > They are running without doing anything in docker-compose
> 
> Dude, they are running locally, you need to test that it works in docker. Where the container_start.sh script is used.

I ran the command for docker as suggested by you. the instances were up. Do we need some another command to test that celery is working or not?

**Comment by iamareebjamal** [comment]
They're running because docker-compose refers to prebuilt image.

Use `docker-compose -f docker-compose-dev.yml up -d`

**Comment by codedsun** [comment]
> They're running because docker-compose refers to prebuilt image.
> 
> Use `docker-compose -f docker-compose-dev.yml up -d`

Ok Will do

**Comment by codedsun** [comment]
@iamareebjamal please do check. Thanks

File: Dockerfile [hunk 1]
```diff
@@ -18,6 +18,7 @@ RUN pip install --prefix=/install --no-warn-script-location -r /requirements.txt
 ####
 
 FROM base
+ARG deployment
```
Comment (iamareebjamal) [inline Dockerfile]: This is not needed

File: docker-compose-dev.yml [hunk 2]
```diff
@@ -2,7 +2,13 @@ version: '3.5'
...
+        context: .
+        args:
+            - deployment=api
 
   celery:
-    build: .
+    build:
+        context: .
+        args:
+            - deployment=celery
```
Comment (iamareebjamal) [inline docker-compose-dev.yml]: Not needed

File: docker-compose.yml [hunk 3]
```diff
@@ -45,15 +45,22 @@ services:
   web:
     <<: *defaults
     container_name: opev-web
+    build:
+        context: .
+        args:
+            - deployment=api
```
Comment (iamareebjamal) [inline docker-compose.yml]: Not needed

File: docker-compose.yml [hunk 4]
```diff
@@ -45,15 +45,22 @@ services:
...
     ports:
       - 8080:8080
 
   celery:
     <<: *defaults
     container_name: opev-celery
+    build:
+        context: .
+        args:
+            - deployment=celery
```
Comment (iamareebjamal) [inline docker-compose.yml]: Not needed

**Comment by codedsun** [comment]
@iamareebjamal I have passed the args from the container and then in the dockerfile overrided the command with a postfix of that arg value

```
CMD ["sh", "scripts/container_start.sh run ${deployment}"]
```

is this approach wrong? but how will I pass value from a container for the command running in the dockerfile. I am new to docker, may be if you can link me with a example or something.

**Comment by iamareebjamal** [comment]
You can override command in docker compose

**Comment by codedsun** [comment]
@iamareebjamal - But for the healtcheck we do want a differentiator for celery container and web container

**Comment by iamareebjamal** [comment]
Which you are already getting in container_start.sh

You have already used that differentiator

**Comment by codedsun** [comment]
@iamareebjamal What I understood is 

adding this in `docker_compose.yml` file

```
 command:
      - /container_start.sh run api 
```
for web container and similar for celery container.

while for healtcheck, we will remove the healthcheck line in `DockerFile` and add that in respective containers in `docker_compose.yml` file?

**Comment by iamareebjamal** [comment]
> we will remove the healthcheck line in DockerFile and add that in respective containers in docker_compose.yml file?

No

**Comment by codedsun** [comment]
@iamareebjamal Then I am still not getting it. 

> Which you are already getting in container_start.sh

Yes as I overrided command in dockercompose.yml and wrote the value their.

But how do we pass that value to DockerFile?

>You have already used that differentiator

I have used in container_start.sh file but I have to do for healthcheck now

**Comment by iamareebjamal** [comment]
Can be done through environment variables. See how other people are doing it

**Comment by codedsun** [comment]
>Can be done through environment variables. See how other people are doing it

Yes we did that, but in this PR i removed that environment variable of `deployment`

**Comment by codedsun** [comment]
@iamareebjamal Check now!

**Comment by codedsun** [comment]
@iamareebjamal Is it correcty now? If yes, then also merge this and I will also update the healthcheck PR. thanks.

**Comment by codedsun** [comment]
@iamareebjamal Also check this if it's good to go?

**Comment by iamareebjamal** [comment]
command should be just `run celery` in celery container and nothing in API, and api is run by default and entrypoint is already set to `container_start.sh`

**Comment by codedsun** [comment]
@iamareebjamal Can you also point me the file where we set the entry point?

**Comment by codedsun** [comment]
@iamareebjamal Also, the issue says something else

```
Instead it should be run via CLI arguments. Like

./container_start.sh run api or ./container_start.sh run celery

```

and `nothing in API, and api is run by default` this will be false as we in this PR we have to specify whether it's `api` or `celery` for starting the script

**Comment by iamareebjamal** [comment]
> Can you also point me the file where we set the entry point?

Dockerfile

**Comment by codedsun** [comment]
@iamareebjamal Also the above comment, 

https://github.com/fossasia/open-event-server/pull/6599#issuecomment-568256774

**Comment by iamareebjamal** [comment]
> this will be false as we in this PR we have to specify whether it's api or celery for starting the script

Then fix it. It should run api by default

**Comment by codedsun** [comment]
Closing this I broke the commit log. 

Opened the new PR for same https://github.com/fossasia/open-event-server/pull/6690

**Pull request closed without merging**